cdap-security/src/main/java/co/cask/cdap/security/guice/SecureStoreServerModule.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.guice;
+
+import co.cask.cdap.api.security.store.SecureStore;
+import co.cask.cdap.api.security.store.SecureStoreManager;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.conf.SConfiguration;
+import co.cask.cdap.security.store.DefaultSecureStoreService;
+import co.cask.cdap.security.store.DummySecureStoreService;
+import co.cask.cdap.security.store.FileSecureStoreService;
+import co.cask.cdap.security.store.SecureStoreService;
+import co.cask.cdap.security.store.SecureStoreUtils;
+import co.cask.cdap.security.store.secretmanager.SecretManagerSecureStoreService;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.PrivateModule;
+import com.google.inject.Provider;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.name.Names;
+
+/**
+ * Server side guice bindings for security store related classes.
+ */
+public class SecureStoreServerModule {
+  public static final String DELEGATE_SECURE_STORE_SERVICE = "delegateSecureStoreService";
+
+  public Module getSecureStoreModule() {
```

Comment:
No need to have a method. Just make `SecureStoreServiceModule extends PrivateModule` and use it directly.